USING_ELEKTRA.md

```diff
@@ -0,0 +1,165 @@
+# Using Elektra
+
+In the new version of lcdproc, [Elektra](https://www.libelektra.org/home) 
+will be used to store and edit configuration settings which are used by lcdproc.
+
+Elektra stores configuration in a global, hierarchical key database. Elektra
+comes with a lot of benefits such as [code generation](https://www.libelektra.org/tools/gen) or
+[advanced GUIs](https://www.libelektra.org/tools/qt-gui). But one major benefit
+ is the possibility to have a 
+ [validation on configuration settings](https://www.libelektra.org/tutorials/validate-configuration)
+ because of configuration specifications.
+
+This tutorial will teach you how to introspect such a specification and how
+to change values accordingly. Be sure to have everything
+installed correctly by first reading through [INSTALL.md](INSTALL.md).
+
+## Basic Concept
+
+Elektra stores values in form of a key value pair in a global, 
+hierarchical key database. You can easily store values with 
+`kdb set <key> <value>` and fetch them via `kdb get <key>`.
+
+Let's take the following example:
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' green
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+> green
+```
+
+Elektra uses the concept of [Namespaces](https://www.libelektra.org/tutorials/namespaces)
+which means that keys are organized similar to directories in a file system. In the case
+of lcdproc, the relevant keys are located under `/sw/lcdproc/<app>/#0/current`
+where `<app>` will either be server configuration `lcdd` or the client configurations
+ `lcdproc`, `lcdvc` and `lcdexec`. So in the upper example we change the background
+ color of the curses driver to green for the server.
+ 
+ Elektra also comes with specifications for lcdproc such as [LCDd-spec.ini](server/specification/LCDd-spec.ini).
+ This guarantees that certain mistakes cannot occur like setting the background color to *greeen* which could
+ potentially crash the application. If you have followed the specification installation process correctly
+ from the [INSTALL.md](INSTALL.md), you now have the full specification [mounted](https://www.libelektra.org/tutorials/mount-configuration-files)
+ under the `spec` namespace. The specification for `/curses/background` background for example
+ tells us that only `red, black, green, yellow, blue, magenta, cyan, white` are valid and that the value given 
+ has to be of type string. Elektra will prevent you to set invalid values which will help reduce errors and misconfiguration.
+  All these data and much more (eg. default values, descriptions, etc.) are saved as 
+ [metadata](https://github.com/ElektraInitiative/libelektra/blob/e82b55f0d3ea4c77f3a4c04fa217084021bd8e4b/doc/dev/metadata.md)
```

Comment:
The link should be to the master branch or to https://www.libelektra.org/devdocu/metadata, in case we change the file. You should also mention METADATA.ini, which describes what all the metakeys do.